decidim-core/lib/decidim/scopable_component.rb

```diff
@@ -16,20 +16,30 @@ module ScopableComponent
       #
       # Returns a boolean.
       def has_subscopes?
-        (scopes_enabled? || participatory_space.scopes_enabled?) && subscopes.any?
+        (scopes_enabled? || parent_scopes_enabled?) && subscopes.any?
       end
 
       # Public: Returns the component Scope
       def scope
-        return participatory_space.scope unless scopes_enabled?
+        return parent_component_or_space.scope unless scopes_enabled?
 
-        participatory_space.scopes.find_by(id: settings.scope_id)
+        parent_component_or_space.scopes.find_by(id: settings.scope_id)
       end
 
       # Returns a boolean.
       def scopes_enabled
         settings.try(:scopes_enabled)
       end
+
+      def parent_component_or_space
+        return parent if parent
```

Comment:
Maybe you could memoize this? Also, I'd write it as a regular `if` instead of using a guard, as the method is very short and having a parent is not a reason to prematurely exit.